src/history/hs.c

```diff
@@ -965,6 +965,14 @@ __wt_find_hs_upd(WT_SESSION_IMPL *session, WT_ITEM *key, uint64_t recno, WT_UPDA
         WT_NOT_READ(modify, true);
         /* Store this so that we don't have to make a special case for the first modify. */
         hs_stop_ts_tmp = hs_stop_ts;
+
+        /*
+         * Resolving update chains of reverse deltas requires the current transaction to look beyond
+         * its current snapshot in certain scenarios. This flag allows us to ignore transaction
+         * visibility checks when reading in order to construct the modify chain, so we can create
+         * the value we expect.
+         */
+        F_SET(session, WT_SESSION_RESOLVING_MODIFY);
         while (upd_type == WT_UPDATE_MODIFY) {
             WT_ERR(__wt_upd_alloc(session, hs_value, upd_type, &mod_upd, NULL));
```

Comment:
We need to clear the flag if there are errors.

Reply:
Yeah @kommiharibabu already mentioned that above. I'll add that in shortly.